enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,621 @@
+will SNAT the client source IP to match that one of the node. This will get
+further NATing as described [here](https://github.com/openshift/microshift/blob/main/docs/contributor/network/ovn_kubernetes_traffic_flows.md#host-to-clusterservice).
+By the time the traffic reaches the NetworkPolicy checks the source IP has
+been replaced with `100.64.0.0/16`, which will not produce a match and traffic
+will get rejected.
+
+To overcome this, all LoadBalancer services in MicroShift that are subject to
+get NetworkPolicy resources to control external traffic must set
+`externalTrafficPolicy: Local`. This setting will not SNAT packets and the
+source IP is preserved until NetworkPolicy checks run.
+
+This setting, however, comes at the expense an even load when having more than
+one node, as traffic will not be sent to another node on the same port. This
+is not an issue in single node, which is MicroShift's use case and design
+principle.
+
+### Workflow Description
+**cluster admin** is a human user responsible for configuring a MicroShift
+cluster.
+
+1. The cluster admin adds specific configuration for the router prior to
+   MicroShift's start.
+2. After MicroShift started, the system will ingest the configuration and setup
+   everything according to it.
+3. The router will enabled/disabled, be exposed on the specified ports, on the
+   specified IPs, allowing only specified IP addresses, all according to the
+   cluster admin provided configuration. This includes the status in the
+   LoadBalancer service type for the router.
+
+### API Extensions
+None
+
+### Implementation Details
+The default router is composed of a bunch of assets part of the MicroShift
+binary. These assets come from the rebase, copied from the original router in
+[cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator).
+There is already a [LoadBalancer service](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/manifests/assets/router/service-cloud.yaml)
+included, which MicroShift does not copy yet. Depending on the options
+configured in MicroShift, this resource will need further customization done
+by MicroShift's start process.
+
+The rest of the changes imply only additions. The LoadBalancer controller needs
+an expansion on its capabilities to include more IPs in the status and log any
+changes to services. All the logic to control NetworkPolicy resources is also
+new.
+
+### Risks and Mitigations
+Some of the features in this enhancement proposal rely on the LoadBalancer
+controller.
+Some of the features rely on NetworkPolicy resources, so the CNI needs
+to support that.
+
+### Drawbacks
+Some of the features depend on a non-agnostic CNI design, which is the
+LoadBalancer controller. A different behavior should be expected if using a
+different CNI.
+
+## Design Details
+### Open Questions
+* Do we want to have defaults for listening ports, or dont open them if there
```

Comment:
If we do not have any ports open by default, upgrades would require configuration changes, right?

What does OpenShift set as the defaults?

Reply:
We need the default behavior after the work for this enhancement to match the default behavior of MicroShift today so that a user upgrading MicroShift does not also have to deploy a new configuration file to retain the behavior they're expecting.